### DISCUSSION TITLE
Open with Vivify

### DISCUSSION THREAD

**Discussion by Tweekism** [jannis-baum/Vivify#122] [Ideas]
Ok, so this might be a stupid question.

But if I wanted to double click on a .md in my file explorer (I'm Linux / Gnome so Nautilus in my case) and have that open rendered in Vivify. What dark magic would be involved in such a quest?

**Comment by jannis-baum** [comment]
This is very cool, I have also been wanting to look into it. Now that you mentioned it I did finally give it a quick look. Seems like it's super platform-dependent. Something that might work for you on Linux are "[desktop entry files](https://wiki.archlinux.org/title/Desktop_entries)". On macOS we probably have to use [Automator](https://support.apple.com/guide/automator/intro-to-automator-aut6e8156d85/mac).

In case you come up with something make sure you post it here for future reference!

**Reply by tuurep** [reply]
Desktop files are way simpler than it may seem, this might do:

`~/.local/share/applications/vivify.desktop`

```
[Desktop Entry]
Name=Vivify
Exec=viv %U
Type=Application
```

**Comment by tuurep** [comment]
Ooh. I've done stuff like this :)

Yes, need to have the correct desktop entry (as above)

But also the mimetype set:

* https://wiki.archlinux.org/title/Xdg-utils#xdg-mime
* Relevant: `xdg-mime default vivify.desktop text/markdown`

^ but this is more of a desktop environment -independent way, and GNOME might have some extra convenient way.

**Comment by Tweekism** [comment]
```
[Desktop Entry]
Name=Vivify
Exec=viv %U
Type=Application
MimeType=text/markdown;
```

Yeah this kinda works, unless the file path has a space in it and then it doesn't. Adding MimeType to the .desktop files means it'll then show up in the right click -> Open With menu. And of course the launcher icon it creates on the app screen does nothing because viv won't launch with no parameters.

All very interesting, I'll look into it more later. 

> ^ but this is more of a desktop environment -independent way, and GNOME might have some extra convenient way.

Yeah this must be one of those rare occasions where Gnome just follows the standard :P

![image](https://github.com/user-attachments/assets/808892c2-ed90-4c4d-9d83-2bfe53840bf1)
